### PR TITLE
TR-29: [WooCommerce] V1.1.0

### PR DESCRIPTION
**Pull request by apetrovici**
= 1.1.0 =
* Renamed GP-API to Unified Commerce Platform (UCP)
* Update PHP-SDK to v2.3.7
* Update UCP 3DS
* Add UCP Sandbox indicator at checkout
* Add UCP separate credentials config for Live and Sandbox

### CODE REVIEW

File: src/Gateways/GpApiGateway.php [hunk 1]
```diff
@@ -136,6 +219,50 @@ protected function add_hooks() {
 		add_action( 'woocommerce_api_globalpayments_threedsecure_challengenotification', array( $this, 'process_threeDSecure_challengeNotification' ) );
 	}
 
+	public function gateway_title( $gateway_title, $gateway_id ) {
```
Comment (slogsdon) [inline src/Gateways/GpApiGateway.php:222]: TBH, this seems like a good enhancement that would help all gateways. We should move this up into `AbstractGateway`.

File: src/Gateways/AbstractGateway.php [hunk 2]
```diff
@@ -386,6 +386,22 @@ public function init_form_fields() {
...
 	}
 
+	/**
+	 * Configuration for the secure payment fields on the client side.
+	 *
+	 * @return array
+	 */
+	protected function secure_payment_fields_config() {
+		try {
+			return $this->get_frontend_gateway_options();
```
[the remark this answers is not in the document]
Comment (apetrovici) [reply src/Gateways/AbstractGateway.php:396]: GP-API and Transit need to do additional requests for configuring the JS lib, get_access_token and create_manifest respectively. Invalid credentials would throw an error.

File: src/Gateways/GpApiGateway.php [hunk 3]
```diff
@@ -63,32 +87,84 @@ public function get_first_line_support_email() {
...
 				'default'     => '',
 			),
-			'app_key' => array(
-				'title'       => __( 'App Key', 'globalpayments-gateway-provider-for-woocommerce' ),
+			'sandbox_app_key' => array(
+				'title'       => __( 'Sandbox App Key', 'globalpayments-gateway-provider-for-woocommerce' ),
+				'type'        => 'password',
+				'description' => '',
+				'default'     => '',
+			),
```
[the remark this answers is not in the document]
Comment (apetrovici) [reply src/Gateways/GpApiGateway.php:110]: This is an improvement proposed by our team, based on the feedback received from onboarding the first merchant with WooCommerce and GP-API.
How will this benefit the user:
- the user will potentially better understand that he needs to have 2 separate access id/key pairs for live/sandbox
- the user will be able to very rapidly switch between live/staging if he needs to test something (as they keys remain there)

File: readme.txt [hunk 4]
```diff
@@ -14,7 +14,7 @@ This extension allows WooCommerce to use the available Global Payments payment g
 - Heartland Portico gateway
 - TSYS Genius gateway
 - TSYS TransIT gateway with TSEP
-- Global Payments API (GP-API) gateway
```
[the remark this answers is not in the document]
Comment (apetrovici) [reply readme.txt:17]: This change is driven by the fact that "GP-API" is an internal term so we should not have that appear anywhere.

**Review by securesubmit-buildmaster — APPROVED** [review]

**Pull request merged**